diff --git a/src/c4/std/span.hpp b/src/c4/std/span.hpp
index 1a0bbdc..3262fad 100644
--- a/src/c4/std/span.hpp
+++ b/src/c4/std/span.hpp
@@ -6,7 +6,12 @@
 #include "c4/language.hpp"
 #endif
 
+#if defined(__has_include)
+#  if __has_include(<span>)
+#    define C4CORE_HAS_STD_SPAN 1
+#  endif
+#endif
 
-#if (C4_CPP >= 20) || defined(__DOXYGEN__)
+#if (C4_CPP >= 20 && defined(C4CORE_HAS_STD_SPAN)) || defined(__DOXYGEN__)
 
 #ifndef C4CORE_SINGLE_HEADER
 #include "c4/substr.hpp"
diff --git a/src/c4/std/span_fwd.hpp b/src/c4/std/span_fwd.hpp
index 25e5bba..43f8eac 100644
--- a/src/c4/std/span_fwd.hpp
+++ b/src/c4/std/span_fwd.hpp
@@ -9,7 +9,12 @@
 #include "c4/language.hpp"
 #endif
 
+#if defined(__has_include)
+#  if __has_include(<span>)
+#    define C4CORE_HAS_STD_SPAN 1
+#  endif
+#endif
 
-#if (C4_CPP >= 20) || defined(__DOXYGEN__)
+#if (C4_CPP >= 20 && defined(C4CORE_HAS_STD_SPAN)) || defined(__DOXYGEN__)
 
 #ifndef C4CORE_SINGLE_HEADER
 #include "c4/substr_fwd.hpp"
